lib/mayaUsd/undo/UsdUndoStateDelegate.cpp

```diff
@@ -327,7 +367,15 @@ void UsdUndoStateDelegate::_OnDeleteSpec(const SdfPath& path, bool inert)
         return;
     }
 
-    SdfDataRefPtr     deletedData = TfCreateRefPtr(new SdfData());
+    SdfDataRefPtr deletedData = TfCreateRefPtr(new SdfData());
+
+    SdfLayer::TraversalFunction copyFunc = std::bind(
```

Comment:
Good point. I always prefer lambda over std::bind to avoid unnecessary overheads and also found it easier to read/write the code. I learned this from Scott's talk long time ago and never had any issues with it.

Also, there are still some ugly std::bind left in UsdUndoStateDelegate.cpp but I don't want to do it now to make more noise.  

addressed ce535e5fbaa0f4389f2469d5caa99345b4f0cebd